source/server/hot_restarting_child.cc

```diff
@@ -172,5 +217,42 @@ void HotRestartingChild::mergeParentStats(Stats::Store& stats_store,
   stat_merger_->mergeStats(stats_proto.counter_deltas(), stats_proto.gauges(), dynamics);
 }
 
+void HotRestartingChild::onSocketEventUdpForwarding() {
+  std::unique_ptr<HotRestartMessage> wrapped_request;
+  while ((wrapped_request =
+              udp_forwarding_rpc_stream_.receiveHotRestartMessage(RpcStream::Blocking::No))) {
+    if (wrapped_request->requestreply_case() == HotRestartMessage::kReply) {
+      ENVOY_LOG(
+          error,
+          "HotRestartMessage reply received on UdpForwarding (we want only requests); ignoring.");
+      continue;
+    }
+    switch (wrapped_request->request().request_case()) {
+    case HotRestartMessage::Request::kForwardedUdpPacket: {
+      const auto& req = wrapped_request->request().forwarded_udp_packet();
+      Network::UdpRecvData packet;
+      packet.addresses_.local_ =
+          Network::Utility::parseInternetAddressAndPortNoThrow(req.local_addr());
```

Comment:
I think if you use `Utility::resolveUrl()` (and format them that way on the sending side) that unix-domain sockets will work, with no additional work needed

Reply:
Done, with adding a new helper function for the inverse of resolveUrl, and a test for that helper.